modules/core/src/main/java/org/locationtech/jts/index/strtree/STRtree.java

```diff
@@ -373,5 +402,91 @@ public Object nearestNeighbour(Envelope env, Object item, ItemDistance itemDist)
           ((ItemBoundable) minPair.getBoundable(1)).getItem()
       };
   }
+  private Object[] nearestNeighbour(BoundablePair initBndPair, double maxDistance, int k) 
```

Comment:
Right, good point!  

You know, in that case these method probably really should have different names (since the return data is different).  So how about calling the new method nearestNeighbourK ?